layer/ro_layer.go

```diff
@@ -120,13 +120,14 @@ func storeLayer(tx MetadataTransaction, layer *roLayer) error {
 	if err := tx.SetCacheID(layer.cacheID); err != nil {
 		return err
 	}
-	if layer.parent != nil {
-		if err := tx.SetParent(layer.parent.chainID); err != nil {
+	// Do not store empty descriptors
+	if layer.descriptor.Digest != "" {
```

Comment:
Definitely felt weird when I wrote it, but convinced myself that a descriptor without a digest is nothing worth storing since it cannot validate content.